api/workOrder/workOrderRouter.js

```diff
@@ -0,0 +1,494 @@
+ *        createdBy:
+ *          type: string
+ *          description: This is a foreign key to Profiles.id (Okta Id)
+ *        assignedTo:
+ *          type: string
+ *          description:
+ *            This is a foreign key to Profiles.id (Okta Id). By default it is
+ *            set to the same user as 'createdBy'
+ *        priority:
+ *          type: object
+ *          properties:
+ *            id:
+ *              type: integer
+ *              description: autoincrements
+ *            name:
+ *              type: string
+ *            color:
+ *              type: string
+ *              description: This is a hex code in the form '#RRGGBB'
+ *        status:
+ *          type: object
+ *          properties:
+ *            id:
+ *              type: integer
+ *              description: autoincrements
+ *            name:
+ *              type: string
+ *      example:
+ *        id: 1
+ *        title: 'Broken Radiator Thermostat'
+ *        description:
+ *          'Radiator Thermo in Apt 224 is broken. Probably needs replaced.'
+ *        company:
+ *            id: 1
+ *            name: 'ACME Property Management'
+ *        property:
+ *            id: 1
+ *            name: 'Lot 24'
+ *            address: '2404 Railroad St, Pittsburgh, PA 15222'
+ *            imageUrl: 'https://bit.ly/3ajfoTV'
+ *            company: 1
+ *        images:
+ *            - id: 1
+ *              url: 'http://path/to/image'
+ *              user: '00ulthapbErVUwVJy4x6'
+ *              workOrder: 1
+ *            - id: 2
+ *              url: 'http://path/to/image'
+ *              user: '00ulthapbErVUwVJy4x6'
+ *              workOrder: 1
+ *        createdBy: '00ulthapbErVUwVJy4x6'
+ *        assignedTo: '00ulthapbErVUwVJy4x6'
+ *        priority:
+ *            id: 2
+ *            name: 'High'
+ *            color: '#F7931B'
+ *        status:
+ *            id: 1
+ *            name: 'Unassigned'
+ *        created_at: '2020-12-15T22:46:05.962Z'
+ *        updated_at: '2020-12-15T22:46:05.962Z'
+ *
+ * /workOrders:
+ *  get:
+ *    description: Returns a list of workOrders associated with the user
+ *    summary: Get a list of workOrders created by or assigned to the user
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - workOrder
+ *    responses:
+ *      200:
+ *        description: array of workOrders
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: array
+ *              items:
+ *                $ref: '#/components/schemas/WorkOrder'
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      403:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ */
+router.get(
+  [
+    '/',
+    '/users?/workOrders?',
+    '/users?/orders?',
+    '/users?/:id/workOrders?',
+    '/users?/:id/orders?',
+    '/company/users?/:id/workOrders?',
+    '/company/users?/:id/orders?',
+    '/company/:companyId/users?/:id/workOrders?',
```

Comment:
Also wondering what this array of endpoints is for. Are you using it for more flexibility, or is it for some other purpose? Also, I'm guessing you have endpoints here with companyId so we have it there for later so we can add additional functionality to searching for a work order by the company id, or am I misunderstanding?

Reply:
That's exactly it. I thought we might want to be able to pull up work orders by company or by property. I may have gotten a bit feature happy with these, though, and created unnecessary endpoints, because I added many after discovering that we can check the request inside the router and determine what the requester is trying to get. Maybe the front-end team can advise me on the ways that they want to be able to access the work orders and I can trim the excess.

Reply:
Cool, thanks for the explanation!